hs/CalendarResponse.hs

```diff
@@ -0,0 +1,230 @@
+    end = endTime dataInOrder
+    dates = getDates session dataInOrder
+
+-- | Creates an event for each course.
+eventsByCourse :: String -> String -> String -> [[String]] -> [[String]] -> [[(String, String)]] -> String -> [[String]]
+eventsByCourse code session sect start end dates timeSystem = map (eventsByDay code session sect timeSystem) (zip' start end dates)
+
+-- | Join three lists together in tuples of three elements
+zip' :: [a] -> [b] -> [c] -> [(a,b,c)]
+zip' _ _ [] = []
+zip' _ [] _ = []
+zip' [] _ _ = []
+zip' (x:xs) (y2:ys) (z:zs) = (x,y2,z):zip' xs ys zs
+
+-- | Creates an event for each day
+eventsByDay :: String -> String -> String -> String -> ([String], [String], [(String, String)]) -> [String]
+eventsByDay code session sect timeSystem (start, end, dates) = if session == "Y" then year else half
+    where
+    year = halfFall ++ halfWinter
+    halfFall = eventsByTime code sect start end (dates !! 0) timeSystem
+    halfWinter = eventsByTime code sect start end (dates !! 1) timeSystem
+    half = eventsByTime code sect start end (dates !! 0) timeSystem
+
+-- | Creates an event for each start/end time.
+eventsByTime :: String -> String -> [String] -> [String] -> (String, String) -> String -> [String]
+eventsByTime code sect start end date timeSystem = 
+    if fst date == "" || snd date == "" then []
+    else map (eventsByDate code sect date timeSystem) (zip start end)
+
+-- | Generates the string that represents the event for each course.
+eventsByDate :: String -> String -> (String, String) -> String -> (String, String) -> String
+eventsByDate code sect (startDate, endDate) timeSystem (start, end) = 
+    "BEGIN:VEVENT\n" ++ "DTSTAMP:" ++ timeSystem ++ 
+    "\nDTSTART;TZID=America/Toronto:" ++ startDate ++ start ++ "\nDTEND;TZID=America/Toronto:" ++
+    startDate ++ end ++ "\nRRULE:FREQ=WEEKLY;UNTIL=" ++ endDate ++ "000000Z" ++ "\nORGANIZER:University of Toronto" ++
+    "\nSUMMARY:" ++ code ++ " " ++ sect ++ "\nCATEGORIES:EDUCATION\n" ++ "END:VEVENT"
+
+-- ** Ordering data
+
+-- | Orders the data for each course.
+orderData :: [Time] -> [[[Double]]]
+orderData courseFields = filter notNull (assignDay $ map timeField courseFields)
+    where
+    notNull field = not $ null field
+
+-- | Organizes the data by the day of the week, assigning a position based on the time field day.
+assignDay :: [[Double]] -> [[[Double]]]
+assignDay lst = [monday, tuesday, wednesday, thursday, friday]
+    where
+    monday = let mon field = field !! 0 == 0.0 in filter mon lst
+    tuesday = let tues field = field !! 0 == 1.0 in filter tues lst
+    wednesday = let wed field = field !! 0 == 2.0 in filter wed lst
+    thursday = let thur field = field !! 0 == 3.0  in filter thur lst
+    friday = let fri field = field !! 0 == 4.0 in filter fri lst
+
+-- ** Start time
+
+-- | Obtains a list of start times for each course.
+startTime :: [[[Double]]] -> [[String]]
+startTime weekFields = map checkTimeStart weekFields
```

Comment:
I'm not entirely certain what this function does, or what `weekFields` is supposed to represent. Can you make this function a bit clearer?